### PR TITLE
Hotfix issue 48

### PR DESCRIPTION
**Pull request by n8layman**
This PR resolves doltr's contribution to issue #48

### CODE REVIEW

**Comment by n8layman** [comment]
Issue #48 was caused by three problems.

1. The net_read_timeout can't be set to 0 error was fixed by the fis server timeout commit to main. However this update wasn't snapshotted and so the renv.lock file didn't reflect the change.

2. Dolt now writes a sql-server.lock file in the /dbPath/.dolt folder to ensure atomic access to local servers. Github actions was using a cached a copy of the dolt directory including the lock file thereby preventing any new processes from starting at all.

3. Dolt now puts processes to sleep when not active. doltr:::dolt_server_find() only looks for 'running' processes when attempting to recycle a server connection. Because the previous connection was writing a lock file, no further processes could start, resulting in target failure.

This PR resolves 1 and 3.

**Comment by noamross** [comment]
Does this mean we should .gitignore the the lockfile in the main repository?

**Comment by n8layman** [comment]
The lock file might be why the checks are failing on `p not found` as well

File: R/server.R [hunk 1]
```diff
@@ -43,7 +43,7 @@ dolt_server <- function(dir = Sys.getenv("DOLT_DIR", "doltdb"),
                         autocommit = TRUE,
                         read_only = FALSE,
                         log_level = "info",
-                        log_out = NULL,
+                        log_out = "proc.log",
```
Comment (noamross) [inline R/server.R:46]: Should this be the default value or was it for testing?

**Comment by n8layman** [comment]
> Does this mean we should .gitignore the the lockfile in the main repository?

I don't think so. We already .gitignore the entire database folder that contains the lockfile. Previously we had cached the local copy of the database via github actions but Emma has now disabled that.

File: R/server.R (hunk 1 above)
Comment (n8layman) [reply R/server.R:46]: > Should this be the default value or was it for testing?

Sorry that was just for testing. Although it was pretty handy to have it write a default log!

**Comment by n8layman** [comment]
Lastly this actually brings up a new issue. We should clean up the lock file during dolt_server_kill/dkill. I haven't thought through how that works with the multi_db = T yet though. My plan is to raise this as a separate issue and get back to dtra-ml for a bit.

File: R/server.R (hunk 1 above)
Comment (n8layman) [reply R/server.R:46]: Okay removed

**Review by noamross — APPROVED** [review]

**Pull request merged**